src/coffee/sync/utils/product.coffee

```diff
@@ -102,14 +102,26 @@ class ProductUtils extends BaseUtils
 
   matchesByIdOrKeyOrSku: (variant1, variant2) ->
     variant1 and variant2 and (
-      (!isNil(variant1.id) and variant1.id == variant2.id) or
+      (!isNil(variant1.sku) and variant1.sku == variant2.sku) or
```

Comment:
Can you use `matchesBySku` function here?

Reply:
please check here 93c1eb0